File: MetaTrader5SDK/Examples/Gateway/TextFeeder.NET/TextFeeder.cs
//+------------------------------------------------------------------+
//|                                          MetaTrader 5 TextFeeder |
//|                             Copyright 2000-2025, MetaQuotes Ltd. |
//|                                               www.metaquotes.net |
//+------------------------------------------------------------------+
namespace TextFeeder.NET
  {
   using MetaQuotes.MT5CommonAPI;
   using System;
   using System.Runtime.InteropServices;
   //+------------------------------------------------------------------+
   //| Text feeder data source .NET example                             |
   //+------------------------------------------------------------------+
   static class CTextFeeder
     {
      //+------------------------------------------------------------------+
      //| Copyright                                                        |
      //+------------------------------------------------------------------+
      public const string Copyright         ="Copyright 2000-2025, MetaQuotes Ltd.";
      //+------------------------------------------------------------------+
      //| Version / build / date of the program                            |
      //+------------------------------------------------------------------+
      public const int    ProgramVersion    =500;
      public const int    ProgramBuild      =101;
      public const string ProgramBuildDate  ="5 Apr 2017";
      public const string ProgramName       ="MetaTrader 5 Text Feeder .NET";
      public const string ProgramDescription="This datafeed allows receiving quotes and news from text files.";
      //+------------------------------------------------------------------+
      //| Common constants                                                 |
      //+------------------------------------------------------------------+
      public const int    STACK_SIZE_COMMON =262144; // size of thread stack
      //+------------------------------------------------------------------+
      //| Starting banner                                                  |
      //+------------------------------------------------------------------+
      static void Banner()
        {
         //--- show the starting banner
         Console.WriteLine("{0} {1} bit {2}.{3}  build {4} ({5})\n" +
                           "Copyright 2000-2025, MetaQuotes Ltd.\n",
                           ProgramName,(IntPtr.Size*8),ProgramVersion/100,
                           ProgramVersion%100,ProgramBuild,ProgramBuildDate);
        }
      //+------------------------------------------------------------------+
      //| Entry point                                                      |
      //+------------------------------------------------------------------+
      [STAThread]
      static int Main(string[] args)
        {
         try
           {
            //--- show the banner
            Banner();
            //---
            using(CMTDatafeedApp datafeed=new CMTDatafeedApp())
              {
               //--- initialize the datafeed
               if(!datafeed.Initialize(args))
                  return(-1);
               //--- start the work of the datafeed
               datafeed.Run();
              }
           }
         catch(Exception ex)
           {
            ExtLogger.Out(EnMTLogCode.MTLogErr, "TextFeeder error [{0}]", ex.Message);
            //--- log may be inaccessible
            Console.WriteLine("TextFeeder error [{0}]",ex.Message);
           }
         //--- exit
         return(0);
        }
     }
  }
//+------------------------------------------------------------------+

File: MetaTrader5SDK/Examples/Gateway/TextFeeder/TextFeeder.cpp
//+------------------------------------------------------------------+
//|                                         MetaTrader 5 Text Feeder |
//|                             Copyright 2000-2026, MetaQuotes Ltd. |
//|                                               www.metaquotes.net |
//+------------------------------------------------------------------+
#include "stdafx.h"
#include "MTDatafeedApp.h"
//+------------------------------------------------------------------+
//| Starting banner                                                  |
//+------------------------------------------------------------------+
void Banner(void)
  {
//--- show the starting banner
   wprintf_s(L"%s %d.%02d  build %d, %s\n"
             L"Copyright 2000-2026, MetaQuotes Ltd.\n",
             ProgramName,ProgramVersion/100,ProgramVersion%100,ProgramBuild,ProgramBuildDate);
  }
//+------------------------------------------------------------------+
//| Entry point                                                      |
//+------------------------------------------------------------------+
int32_t wmain(int32_t argc,wchar_t** argv)
  {
   CMTDatafeedApp datafeed;
//--- show the banner
   Banner();
//--- initialize the datafeed
   if(!datafeed.Initialize(argc,argv))
      return(-1);
//--- start the work of the datafeed
   datafeed.Run();
//--- complete the work of the datafeed
   datafeed.Shutdown();
//--- exit
   return(0);
  }
//+------------------------------------------------------------------+
